int32 ID
int32 prevID
int32 nextID
# the length of the segment
float64 length
# start flag and end flag for robot passing
bool start
bool end
